File: gremlin-scala/src/test/scala/com/tinkerpop/gremlin/scala/filter/HasNotStepTest.scala
package com.tinkerpop.gremlin.scala.filter

import com.tinkerpop.blueprints.pgm.impls.tg.TinkerGraphFactory
import com.tinkerpop.gremlin.test.ComplianceTest
import com.tinkerpop.gremlin.scala._
import com.tinkerpop.gremlin.Tokens.T

/**
 * @author Marko A. Rodriguez (http://markorodriguez.com)
 */

class HasNotStepTest extends com.tinkerpop.gremlin.test.filter.HasNotStepTest {

  val g = TinkerGraphFactory.createTinkerGraph();

  override def testCompliance() {
    ComplianceTest.testCompliance(this.getClass)
  }

  def test_g_V_hasNotXname_markoX() {
    super.test_g_V_hasNotXname_markoX(g.V.hasNot("name", "marko"));
  }

  def test_g_V_hasNotXname_blahX() {
    super.test_g_V_hasNotXname_blahX(g.V.hasNot("name", "blah"));
  }

  def test_g_V_hasNotXblah_nullX() {
    super.test_g_V_hasNotXblah_nullX(g.V.hasNot("blah", null));
  }

  def test_g_V_hasNotXage_gt_32X() {
    super.test_g_V_hasNotXage_gt_32X(g.V.hasNot("age", T.gt, 32));
  }
}

File: gremlin-java/src/test/java/com/tinkerpop/gremlin/java/filter/HasNotStepTest.java
package com.tinkerpop.gremlin.java.filter;

import com.tinkerpop.blueprints.Graph;
import com.tinkerpop.blueprints.impls.tg.TinkerGraphFactory;
import com.tinkerpop.gremlin.java.GremlinPipeline;
import com.tinkerpop.gremlin.test.ComplianceTest;

/**
 * @author Marko A. Rodriguez (http://markorodriguez.com)
 */
public class HasNotStepTest extends com.tinkerpop.gremlin.test.filter.HasNotStepTest {

    Graph g = TinkerGraphFactory.createTinkerGraph();

    public void testCompliance() {
        ComplianceTest.testCompliance(this.getClass());
    }

    public void test_g_V_hasNotXname_markoX() {
        super.test_g_V_hasNotXname_markoX(new GremlinPipeline(g).V().hasNot("name", "marko"));
        super.test_g_V_hasNotXname_markoX(new GremlinPipeline(g).optimize(false).V().hasNot("name", "marko"));
    }

    public void test_g_V_hasNotXname_blahX() {
        super.test_g_V_hasNotXname_blahX(new GremlinPipeline(g.getVertices()).hasNot("name", "blah"));
        super.test_g_V_hasNotXname_blahX(new GremlinPipeline(g).optimize(false).V().hasNot("name", "blah"));
    }

    public void test_g_V_hasNotXblahX() {
        super.test_g_V_hasNotXblahX(new GremlinPipeline(g).V().hasNot("blah"));
        super.test_g_V_hasNotXblahX(new GremlinPipeline(g.getVertices()).optimize(false).hasNot("blah"));
    }
}